src/contracts/Staking.sol

```diff
@@ -113,13 +155,92 @@ contract Staking is Ownable {
         });
     }
 
+    /**
+        @notice claim TOKE from Tokemak
+        @param _claimAddress address
+        @param _amount uint
+        @param _v uint
+        @param _r bytes
+        @param _s bytes
+     */
+    function claimFromTokemak(
+        address _claimAddress,
+        uint256 _amount,
+        uint8 _v,
+        bytes32 _r,
+        bytes32 _s
+    ) public onlyManager {
+        ITokeReward tokeRewardContract = ITokeReward(tokeReward);
+        ITokeRewardHash iTokeRewardHash = ITokeRewardHash(tokeRewardHash);
+
+        uint256 latestCycleIndex = iTokeRewardHash.latestCycleIndex();
+        Recipient memory recipient = Recipient({
+            chainId: 1,
+            cycle: latestCycleIndex,
+            wallet: address(this),
+            amount: _amount
+        });
+        tokeRewardContract.claim(recipient, _v, _r, _s);
```

Comment:
Hmm, this seems potentially dangerous.  How we do know what we are claiming from the rewards contract, no amount is passed or checked and then we send FOX to the _claimAddress, unclear on this one.

Reply:
As per my comment above I think these can be separated out and it will resolve any issues.  We have to pass in the amount to claim and we get that amount through the Tokemak ipfs

Reply:
But i am still not clear on why we are transferring FOX and not TOKE, please advise

Reply:
That is a great point, good catch! Should definitely be TOKE